src/Integration.UnitTests/Binding/CFamilyProjectBinderTests.cs

```diff
@@ -0,0 +1,145 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO.Abstractions;
+using System.Threading;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration.Binding;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Binding
+{
+    [TestClass]
+    public class CFamilyProjectBinderTests
+    {
+        private Mock<IServiceProvider> serviceProvider;
+        private Mock<ISolutionRuleSetsInformationProvider> solutionRuleSetsInformationProviderMock;
+        private Mock<IFileSystem> fileSystemMock;
+        private TestLogger logger;
+        private CFamilyProjectBinder testSubject;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            solutionRuleSetsInformationProviderMock = new Mock<ISolutionRuleSetsInformationProvider>();
+
+            serviceProvider = new Mock<IServiceProvider>();
+            serviceProvider
+                .Setup(x => x.GetService(typeof(ISolutionRuleSetsInformationProvider)))
+                .Returns(solutionRuleSetsInformationProviderMock.Object);
+
+            fileSystemMock = new Mock<IFileSystem>();
+            logger = new TestLogger();
+            testSubject = new CFamilyProjectBinder(serviceProvider.Object, logger, fileSystemMock.Object);
+        }
+
+        [TestMethod]
+        public void Ctor_NullServiceProvider_ArgumentNullException()
+        {
+            Action act = () => new CFamilyProjectBinder(null, Mock.Of<ILogger>(), Mock.Of<IFileSystem>());
+
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("serviceProvider");
+        }
+
+        [TestMethod]
+        public void Ctor_NullLogger_ArgumentNullException()
+        {
+            Action act = () => new CFamilyProjectBinder(serviceProvider.Object, null, Mock.Of<IFileSystem>());
+
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("logger");
+        }
+
+        [TestMethod]
+        public void Ctor_NullFileSystem_ArgumentNullException()
+        {
+            Action act = () => new CFamilyProjectBinder(serviceProvider.Object, Mock.Of<ILogger>(), null);
+
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("fileSystem");
+        }
+
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void IsBound_ProjectHasOneLanguage(bool configFileExists)
+        {
+            var projectMock = new ProjectMock("c:\\test.csproj");
+            projectMock.SetCSProjectKind();
+
+            solutionRuleSetsInformationProviderMock
+                .Setup(x => 
+                    x.CalculateSolutionSonarQubeRuleSetFilePath("key", Language.CSharp, SonarLintMode.Connected))
```

Comment:
There's something wrong, either in the test or in the product code. The `CFamilyProjectBinder` should only be handling CFamily languages - it shouldn't be checking whether config files exist for non-CFamily languages.

It doesn't need to block this commit, but there's definitely more work to do on how "IsBound" is implemented i.e. a TODO that needs to be captured somewhere.

Reply:
Ah, I see your point.
It returns back to the discussion we had this morning regarding the language per project handling: CFamilyProjectBinder doesn't know how many languages the project has, while in practice there would always be exactly 2 (cpp+c). So I wanted to write a test that mimics the fact that there is only one language per project.
But maybe I should remove this test entirely, 1) because it won't happen in reality, and 2) because it's confusing.

Reply:
I think it's confusing because it's raising some design questions we haven't addressed e.g.
* what should a project binder do if it doesn't care about any of the languages in the project?
* should CFamily project binder know that _at present_ if a project supports `C` then it also supports `C++`?

Having a test is useful because it forces us to think about what the behaviour should be, and acts as record of those decisions.

It's up to you whether to remove this test for now, although I agree with your reason for adding it - I think the binder should handle the presence of only one language. The bit that is causing me a problem is that the fact that test is using a C# project!

We can add more tests later when we've decided what the behaviour should be.